allennlp/training/util.py

```diff
@@ -228,24 +229,29 @@ def create_serialization_dir(
                                      "does not exist.  There is nothing to recover from.")
         os.makedirs(serialization_dir, exist_ok=True)
 
-def data_parallel(batch, model: Model, cuda_devices: List) -> Dict[str, torch.Tensor]:
+def data_parallel(batch_group: List[TensorDict],
+                  model: Model,
+                  cuda_devices: List) -> Dict[str, torch.Tensor]:
     """
     Performs a forward pass using multiple GPUs.  This is a simplification
     of torch.nn.parallel.data_parallel to support the allennlp model
     interface.
     """
-    inputs, module_kwargs = scatter_kwargs((), batch, cuda_devices, 0)
+    assert len(batch_group) <= len(cuda_devices)
 
-    used_device_ids = cuda_devices[:len(inputs)]
+    inputs = [()] * len(batch_group)
```

Comment:
`inputs` is supposed to be a list of empty tuples?  This never gets updated before getting passed to `parallel_apply`.

Reply:
Added a comment to clarify. You can see that `()` was passed to the old `scatter_kwargs` as well.

```
# We pass all our arguments as kwargs. Create a list of empty tuples of the
# correct shape to serve as (non-existent) positional arguments.
```